packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3292,6 +3294,86 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  /// Handles [ScrollIntent] by scrolling the [Scrollable] inside of
+  /// [EditableText].
+  void _scroll(ScrollIntent intent) {
+    if (intent.type != ScrollIncrementType.page) {
+      return;
+    }
+
+    final ScrollPosition position = _scrollController.position;
+    if (widget.maxLines == 1) {
+      _scrollController.jumpTo(position.maxScrollExtent);
+      return;
+    }
+
+    final ScrollableState? state = _scrollableKey.currentState as ScrollableState?;
+    final double increment = ScrollAction.getIncrement(state!, intent);
+    final double destination = (position.pixels + increment).clamp(
+      position.minScrollExtent,
+      position.maxScrollExtent,
+    );
+    if (destination == position.pixels) {
+      return;
+    }
+    _scrollController.jumpTo(destination);
+  }
+
+  /// Extend the selection down by page if the `forward` parameter is true, or
+  /// up by page otherwise.
+  void _extendSelectionByPage(ExtendSelectionByPageIntent intent) {
+    if (widget.maxLines == 1) {
+      return;
+    }
+
+    final TextSelection nextSelection;
+    final Rect extentRect = renderEditable.getLocalRectForCaret(
+      _value.selection.extent,
+    );
+    final ScrollableState? state = _scrollableKey.currentState as ScrollableState?;
+    final double increment = ScrollAction.calculateScrollIncrement(
```

Comment:
Since this is only used for vertically scrolled / multi-line fields, I wonder if using state.renderEditable.size.height instead of adding ScrollAction.calculateScrollIncrement() would be sufficient?

Reply:
> I wonder if using state.renderEditable.size.height instead of adding ScrollAction.calculateScrollIncrement() would be sufficient

It could be for this specific use case, but for consistency across the framework, we should probably establish a pattern. :)

Reply:
Now only one method is exposed, `getDirectionalIncrement`.  It just uses a ScrollIntent like usual, which I create here.